### PR TITLE
Add CI runtime with short blocktime

### PR DESCRIPTION
**Pull request by polymath-eric**
Adds in a new CI runtime.

This runtime has short block times so it can run quickly for automated tests.

### CODE REVIEW

File: src/chain_spec.rs [hunk 1]
```diff
@@ -1174,3 +1174,137 @@ pub mod mainnet {
...
+    session_keys!();
+
+    fn genesis(
+        initial_authorities: Vec<InitialAuth>,
+        root_key: AccountId,
+        _enable_println: bool,
+        treasury_bridge_lock: BridgeLockId,
+        key_bridge_locks: Vec<BridgeLockId>,
+    ) -> rt::runtime::GenesisConfig {
+        let (identities, stakers, complete_txs) = genesis_processed_data(
```
Comment (adamdossa) [inline src/chain_spec.rs:1193]: You may want to use `dev_genesis_processed_data` instead if you want to be closer to our usual `dev` chains. The difference is in the initial set of identities etc. which are set up on genesis.

**Review by adamdossa — APPROVED** [review]

File: src/chain_spec.rs (hunk 1 above)
Comment (raycar5) [reply src/chain_spec.rs:1193]: nope, it should be the same as `testnet-dev` which is what the current integration tests are based on and `testnet-dev` uses genesis_processed_data

**Review by raycar5 — CHANGES_REQUESTED** [review]
I'd swap over the integration tests in this pr to the ci runtime to make sure it runs properly in ci, you can do it in `environment.config.js`

**Comment by Centril** [comment]
Is this going to be used for CI on this repo?

Should we try to deduplicate with the develop runtime?

File: scripts/cli/test.sh [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 
 set -e
+npm run build
```
Comment (polymath-eric) [inline scripts/cli/test.sh:4]: I had failing tests and didn't realize I had to rebuild them.

This runs quickly and stops old versions from running.

File: scripts/cli/src/tests/11_A_settlement.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ async function main(): Promise<void> {
   await settlement.affirmInstruction(bob, intructionCounterAB, bobDid, 0);
 
   // Wait for settlement to be executed - happens in the next block
-  await init.sleep(15000);
+  await init.sleep(1500);
```
Comment (polymath-eric) [inline scripts/cli/src/tests/11_A_settlement.ts:43]: Faster chain means less waiting

**Comment by polymath-eric** [comment]
> Is this going to be used for CI on this repo?
> 
> Should we try to deduplicate with the develop runtime?

This is going to be used for CI for multiple repos. There is [polymesh-local](https://github.com/PolymathNetwork/polymesh-local) which will grab this CI version of the chain, along with Subquery + tooling, so we have an environment to run e2e tests against.

I'd be happy to hear any ideas to deduplicate code here.

**Comment by Neopallium** [comment]
Should be able to just add a feature to switch to shorter blocktimes.

**Comment by raycar5** [comment]
then you'd need to recompile the chain or set up a new build

**Comment by Neopallium** [comment]
What do we use the `testnet` chain for?

**Comment by polymath-eric** [comment]
`testnet` is for chains like Alcyone

**Comment by Neopallium** [comment]
FYI, I tried to dedup `testnet` and `ci` chain runtimes.  Each runtime is compiled twice (once with `std` and once without `std` for wasm version) using a `build.rs` script from substrate.  That makes it very difficult to re-use `polymesh-runtime-testnet` with the shorter blocktime.

The only option to dedup the code would be to use a feature flag that is only enabled for the CI pipeline.
How is the node binary for `Alcyone` built and deployed? @adamdossa

**Comment by adamdossa** [comment]
> How is the node binary for `Alcyone` built and deployed? @adamdossa
@Neopallium 
Not sure I follow exactly - the binary is the same for all of the chains, they just have a different chain-spec and WASM.

Concourse takes care of building the official releases, triggered by commits to `alcyone` branch.

**Review by satyamakgec — APPROVED** [review]

**Comment by Neopallium** [comment]
> > How is the node binary for `Alcyone` built and deployed? @adamdossa
> > @Neopallium
> > Not sure I follow exactly - the binary is the same for all of the chains, they just have a different chain-spec and WASM.
> 
> Concourse takes care of building the official releases, triggered by commits to `alcyone` branch.

To avoid a lot of duplicate code (`polymesh-runtime-testnet` is the same as `polymesh-runtime-ci` except for the blocktime), we would like to use a feature flag (i.e. 'short-blocktime`) during the CI pipeline run.  But if the deployed (official release) binary comes from our CI pipeline, then we can't turn on that feature flag because it would affect the official release.

We could possibly add a second `build` stage to build a special binary for the CI integration tests.

**Comment by raycar5** [comment]
we specifically want the ci runtime to be included in the releases so that polymesh-local can grab it from there, I think the only solution is code duplication or resorting to macros, and we don't really have time to resort to macros here

**Review by raycar5 — APPROVED** [review]

**Pull request merged**